### PR TITLE
Device groups

### PR DESCRIPTION
**Pull request by murrant**
Implement Device Groups.
The format differs from V1, it does convert V1 filters for use (may need more testing), but it only saves v2 style (SQL).  Eloquent/Fluent should protect us from sql injection, but that may need more research.

Uses http://querybuilder.js.org/ to edit/create rules.  Very configurable and if we customize are filter, we could have an even nicer input, but I left that for now.

Various other cleanups in here, notably modal cleanups and better js/css includes.

### CODE REVIEW

**Comment by laf** [comment]
Running as test now.

File: resources/views/devices/group-list.blade.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    @parent
+    <link href="{{ url('js/plugins/querybuilder/query-builder.default.min.css') }}" rel="stylesheet"/>
+    <link href="{{ url('js/plugins/select2/select2.min.css') }}" rel="stylesheet"/>
+
+@endsection
+
+@section('pagejs')
+    @parent
+    <script src="{{ url('js/core/modals.js') }}"></script>
+    <script src="{{ url('js/plugins/querybuilder/query-builder.standalone.js') }}"></script>
```
Comment (laf) [inline resources/views/devices/group-list.blade.php]: Should be:

<script src="{{ url('js/plugins/querybuilder/query-builder.standalone.min.js') }}"></script>

Comment (murrant) [inline resources/views/devices/group-list.blade.php]: Oops, I removed the non-min version and didn't notice I included this.

**Comment by murrant** [comment]
@librenms/reviewers I'm unsure which option I should pursue for storage of the patterns.  The complication is that the query builder is more powerful than the current editor.  It allows for grouping for example.

I've initially coded option 2 in this PR (if you want to play with it).  
1. Disable the nested grouping in jQuery-QueryBuilder and store patterns in v1 format. Currently, this would allow only one operator (AND or OR) for all rules.  This issue would fix that: https://github.com/mistic100/jQuery-QueryBuilder/issues/301
2. Store the where query as is in the database, only need to expand macros before updating the group associations.
3. Store in a third format that allows for easy sql validation.  The pattern with question marks to denote values and a json encoded array of the values.  This would require more work parsing the patterns, but allow for easier avoidance of sql injection in v1 (injection is already prevented by Laravel for v2)

For options 2 & 3 I would update the v1 code to work wih the newer style patterns.

**Comment by laf** [comment]
How about we make a decision on v2 not being directly compatible and a migration script is provided?

I think with all the best will in the world we will struggle to shoe horn some of the bad decisions from the past into what should be a clean slate.

**Pull request merged**